js/damage.js

```diff
@@ -120,9 +120,13 @@ function getDamageResult(attacker, defender, move, field) {
     if (typeEffectiveness === 0) {
         return {"damage":[0], "description":buildDescription(description)};
     }
-    if (move.name === "Sky Drop" && (defender.type1 === "Flying" || defender.type2 === "Flying" || (gen >= 6 && defender.weight >= 200))) {
+    if (move.name === "Sky Drop" && (defender.type1 === "Flying" || defender.type2 === "Flying" || (gen >= 6 && defender.weight >= 200) || field.isGravity)) {
         return {"damage":[0], "description":buildDescription(description)};
     }
+    if (move.name === "Synchronoise" &&
```

Comment:
No, what you did is fine. That would only check `type1` against `type1` since it always exists.

What I'm wondering is why the Sky Drop change? None of the other moves disabled by Gravity are covered either. If we want to do that, might as well flag them in the movedex and check it only once somewhere else.